### DISCUSSION TITLE
Cant get namespaces to work, need help

### DISCUSSION THREAD

**Discussion by TimurFidarov** [phalcon/cphalcon#15315] [Q&A] [answered]
Hi, i'm new to phalcon. I've been trying to follow documentation for namespaces, but at this point i don't know what to do next.

So the problem is that when i try to use $loader->registerNamespaces() instead of registerDirs(), the dispatcher cant find my controllers. 
Here my index.php file

```
<?php

use Phalcon\Di\FactoryDefault;
use Phalcon\Loader;
use Phalcon\Mvc\View;
use Phalcon\Mvc\Application;
use Phalcon\Url;
use Phalcon\Db\Adapter\Pdo\Mysql;

// Define some absolute path constants to aid in locating resources
define('BASE_PATH', dirname(__DIR__));
define('APP_PATH', BASE_PATH . '/app');

// Register an autoloader
$loader = new Loader();

//$loader->registerDirs(
//    [
//        APP_PATH . '/controllers/',
//        APP_PATH . '/models/',
//        APP_PATH . '/modules/'
//    ]
//);

$loader->registerNamespaces(
    [
     'App\Controllers' =>   APP_PATH . '/controllers/'
    ]
);

$loader->register();


$container = new FactoryDefault();

$container->set(
    'view',
    function () {
        $view = new View();
        $view->setViewsDir(APP_PATH . '/views/');
        return $view;
    }
);

$container->set(
    'db',
    function () {
        return new Mysql(
            [
                'host'     => 'localhost',
                'username' => 'sammy',
                'password' => 'password',
                'dbname'   => 'learning',
            ]
        );
    }
);

$container->set(
    'url',
    function () {
        $url = new Url();
        $url->setBaseUri('/');
        return $url;
    }
);

$application = new Application($container);


    // Handle the request
    $response = $application->handle(
        $_SERVER["REQUEST_URI"]
    );

    $response->send();

```

and here my IndexController.php file 

```
<?php
declare(strict_types=1);
namespace App\Controllers;

use Phalcon\Http\Request;
use \Phalcon\Mvc\Controller;
use Phalcon\Http\Message\Uri;

class IndexController extends Controller
{

    public function indexAction()
    {
        return 'hi';
    }
}
```

Here is my directory structure: 
![image](https://user-images.githubusercontent.com/57711157/109275655-14eb7e00-7826-11eb-82dd-f98fb8ff7b1d.png)

My first thought was that the problem is with the path, but as u can see i define same path for registerDir function and it works. Anyway i get error, that says: "IndexController handler class cannot be loaded" and i don't have a clue why.

**Comment by Jeckerson** [comment]
Hello, you can check this x2 demo projects as reference.

* https://github.com/phalcon/vokuro
* https://github.com/phalcon/invo

**Comment by bahadirbirsoz** [comment, marked as answer]
It's been a while but I wanted to add a quick answer for future reference. 

Your controllers' namespace is `App/Controllers` and it is located in `app/controllers` directory. 
Similarly, your models will have `App/Models` namespace and will be located in `app/models`. 

In this case all you need to register is the `App` Namespace to `APP_PATH`. Each namespace that begins with `App\` will be mapped with directory names. In other words, each class file under app directory and its subdirectories will be expected to have relatively mapped directory names as namespace (of course camel case).  

```
$loader->registerNamespaces([
    'App' => APP_PATH,
])->register();
```
